### ISSUE TITLE
Why are there so many operations that I can't perform as a call/function?

### ISSUE THREAD

**Issue by hotmatrixx** [binary-com/binary-bot#2357]
basically anything that has a built in list - candle lists, tick lists, along with a few other things, can't be put into functions.

I like functions - I use them like subroutines in OOP, and with the lack of //REM functionality they make keeping track of code snippets very tidy.

but there are many little things that will not let me use them; from Candle and Tick lists to timers to  contract results to "Purchase Rise" and "Trade Again"

There is no need for these to be illegal inside functions; in fact it reduces their usefulness.

![Capture](https://user-images.githubusercontent.com/49715381/72966305-7a02c200-3e23-11ea-8f82-830648235730.PNG)

**Comment by Senjamris** [comment]
Bro I know this is irrelevant but I need this bot called '' Forex-SmartFX bot". If u have this bot then plz send it to me. If possible plz add some strategy in that bot like ema or rsi whatever to get more accurate.

**Comment by aaron-binary** [comment]
Hey @hotmatrixx,

You're absolutely right, these restrictions don't make sense when the functions are called within scopes that allow for these block's usages. We'll work on updating this.

Thanks for your feedback,
Aaron

**Comment by hotmatrixx** [comment]
@anton-binary 
That's absolutely brilliant, thanks!

The main one that frustrates me is that if the payout changes, I can only find the new payout and change the Stake once I have won/lost another round - 

Ideally, I'd want to be able to check the payout value and adjust my stake before placing my defined Trade Options; 

More specifically, I'd like to be able to run "Block 2" for checking my 'required market conditions', then once they all fit, THEN calculate my stake based on the current payout, THEN place my position.

This would also solve problems for people who want to place trades based on previous ticks for those who are using 'digits matches/differs' strategies.

But,that might require a fair bit of rework in the bot and could allow for lots of broken logic if people don't understand the bot as intimately as myself, binaryexmachina, you, and a couple of others.

So basically, I'd ideally like a slightly different workflow diagram.
I know this means a lot of work, but with Deriv.com it might be a good time to look into it?
![Untitled](https://user-images.githubusercontent.com/49715381/73042867-b7209000-3ec7-11ea-8101-222a869c25ad.png)
